components/dashboard/src/projects/Prebuilds.tsx

```diff
@@ -193,13 +193,13 @@ export default function () {
                     </ItemField>
                     <ItemField className="flex items-center">
                         <div className="truncate">
-                            <div className="text-base text-gray-500 dark:text-gray-50 font-medium mb-1 truncate">{shortCommitMessage(p.info.changeTitle)}</div>
+                            <div className="text-base text-gray-500 dark:text-gray-50 font-medium mb-1 truncate" title={shortCommitMessage(p.info.changeTitle)}>{shortCommitMessage(p.info.changeTitle)}</div>
                             <p>{p.info.changeAuthorAvatar && <img className="rounded-full w-4 h-4 inline-block align-text-bottom mr-2" src={p.info.changeAuthorAvatar || ''} alt={p.info.changeAuthor} />}Authored {formatDate(p.info.changeDate)} · {p.info.changeHash?.substring(0, 8)}</p>
                         </div>
                     </ItemField>
-                    <ItemField className="flex items-center">
-                        <div className="flex space-x-2">
-                            <span className="font-medium text-gray-500 dark:text-gray-50">{p.info.branch}</span>
+                    <ItemField className="flex" shouldCenterVertically={false}>
+                        <div className="flex space-x-2 truncate">
+                            <span className="font-medium text-gray-500 dark:text-gray-50 truncate" title={p.info.branch}>{p.info.branch}</span>
```

Comment:
Also looks good to me!

One small nit: I think it's an anti-pattern to add a `truncate` class to both an element and its container. In theory, only one should be required. Maybe you can remove the `truncate` class from the container? (Haven't actually tested this -- if it doesn't work, please don't worry about this comment 🙏)

Reply:
Again, 100% agree.

When I was starting to work on this, I think I found an old commit adding the same thing to another piece of the app (maybe it was just a few lines above these, I can't remember) and I found the double `truncate` to be weird... but it turns out it is needed!  It doesn't work without.

That's because the CSS ellipsis needs the parent to have certain properties in order to work as expected, and the `truncate` class happens to also add those. So at the end of the day it's a hack, a workaround, that I found in the codebase and applied here.

Here's my proposal: I'll create a new `TruncatedText` component in `components/dashboard/src/components` which will:
- provide the necessary CSS "environment" for `truncate` to work
- use `truncate` on its `children` text
- accept a `title` prop

This will result in another refactoring over 38 files 😅 but, hey, I'll do it 💪.

Thoughts?
Also, since this new double `truncate` is perfectly in line with what's already around there, maybe I could open another issue / PR and work on it as my next task?

Reply:
Aha, many thanks for explaining why the double `truncate` is needed @trumbitta.

Maybe the underlying reason is that Tailwind's `truncate` class is actually a combination of several CSS classes: https://tailwindcss.com/docs/text-overflow

If possible, I think I'd prefer a refactoring that makes our code always use truncate class(es) properly, instead of creating a brand new React component to solve this, but I could be wrong here.

> Also, since this new double `truncate` is perfectly in line with what's already around there, maybe I could open another issue / PR and work on it as my next task?

Agreed. 👍 I think you could indeed open an issue about unifying how we truncate text, and we could move this discussion there. Thanks for caring about this!